test/topology/test_tablets.py

```diff
@@ -65,10 +68,9 @@ async def test_tablet_metadata_propagates_with_schema_changes_in_snapshot_mode(m
 
     # Check that after rolling restart the tablet metadata is still there
     for s in servers:
-        manager.server_restart(s, wait_others=2)
+        await manager.server_restart(s.server_id, wait_others=2)
```

Comment:
Well, looks like the test was completely wrong before and had no right to work, this is much more than fixing flakiness.

For the future I recommend using mypy https://mypy.readthedocs.io/en/stable/. All the `ManagerClient` etc. (generally framework) functions provide type hints.

Maybe I should add mypy checking to CI as part of the quality week...

Reply:
How many flaky tests do you remember from the past that mypy would have caught? One? Five?

I'm personally against any friction that will cause developers not to develop tests because it's annoying, or spend time on adding silly comments and type hints to already-correct code.

Reply:
Static type checking removes friction.

Reply:
Also, perhaps there's a misunderstanding. When developing tests, for `mypy` to provide value here:
```
        await manager.server_restart(s.server_id, wait_others=2)
```
you don't need to write **any** additional comments and type hints. The type hints are coming from the test framework. You only run `mypy` (preferably your editor/IDE is configured to run it continuously and provide feedback in real-time).